### PR TITLE
Add missing & Remove duplicated files in CMakeLists.txt

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PR aims to

- Add missing header files in CMakeLists.txt
- Sort alphabetically the lists of headers and sources in CMakeLists.txt for a better organization & avoiding duplications
- Remove duplicated header/sources files in CMakeLists.txt
- Some additional cosmetic actions mainly related to adding and removing spaces in CMakeLists.txt files

### CODE REVIEW

File: src/coreComponents/dataRepository/CMakeLists.txt [hunk 1]
```diff
@@ -44,14 +46,14 @@ if( ENABLE_PYGEOSX )
...
-blt_add_library( NAME                  dataRepository
-                 SOURCES               ${dataRepository_sources}
-                 HEADERS               ${dataRepository_headers}
-                 DEPENDS_ON            ${dependencyList}
-                 OBJECT                ${GEOSX_BUILD_OBJ_LIBS}
+blt_add_library( NAME       dataRepository
+                 SOURCES    ${dataRepository_sources}
+                 HEADERS    ${dataRepository_headers}
+                 DEPENDS_ON ${dependencyList}
+                 OBJECT     {GEOSX_BUILD_OBJ_LIBS}
```
Comment (klevzoff) [inline src/coreComponents/dataRepository/CMakeLists.txt:53]: This typo is the likely reason for CI build failures:
```suggestion
                 OBJECT     ${GEOSX_BUILD_OBJ_LIBS}
```

File: src/coreComponents/functions/CMakeLists.txt [hunk 2]
```diff
@@ -38,13 +36,13 @@ if ( ENABLE_CUDA )
...
-blt_add_library( NAME                  functions
-                 SOURCES               ${functions_sources}
-                 HEADERS               ${functions_headers}
-                 DEPENDS_ON            ${dependencyList}
-                 OBJECT                ${GEOSX_BUILD_OBJ_LIBS}
+blt_add_library( NAME       functions
+                 SOURCES    ${functions_sources}
+                 HEADERS    ${functions_headers}
+                 DEPENDS_ON ${dependencyList}
+                 OBJECT     {GEOSX_BUILD_OBJ_LIBS}
```
Comment (klevzoff) [inline src/coreComponents/functions/CMakeLists.txt:43]: Another one here:
```suggestion
                 OBJECT     ${GEOSX_BUILD_OBJ_LIBS}
```

**Review by bmhan12 — COMMENTED** [review]
Nice cleanup! One change I see is that `GEOSX_PTP` was refactored and removed as a submodule in PR #1498, and this PR is trying to add it back in as a submodule.

**Comment by sytuannguyen** [comment]
> This PR is trying to add it back in as a submodule.

Thank! This is a mistake, I will remove it

**Review by bmhan12 — APPROVED** [review]

**Pull request merged**